src/main/java/uk/ac/cam/cl/dtg/segue/api/AdminFacade.java

```diff
@@ -281,6 +294,112 @@ public synchronized Response modifyUsersRole(@Context final HttpServletRequest r
     return Response.ok().build();
   }
 
+  /**
+   * This method will allow users to have their teacher pending status mass changed.
+   *
+   * @param request - to help determine access rights.
+   * @param status    - new teacher pending status.
+   * @param userIds - a list of user ids to change en-mass
+   * @return Success shown by returning an ok response
+   */
+  @POST
+  @Path("/users/teacher_pending/{status}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Consumes(MediaType.APPLICATION_JSON)
+  public synchronized Response modifyUsersTeacherPendingStatus(@Context final HttpServletRequest request,
+                                                               @PathParam("status") final Boolean status,
+                                                               final List<Long> userIds) {
+
+    try {
+      RegisteredUserDTO requestingUser = userManager.getCurrentRegisteredUser(request);
+      if (!isUserAnAdminOrEventManager(userManager, requestingUser)) {
+        return new SegueErrorResponse(Status.FORBIDDEN, ACCESS_DENIED_MESSAGE)
+            .toResponse();
+      }
+
+      Map<String, List<Long>> failedUpdates = new HashMap<>();
```

Comment:
If I gave each user their own error (if any happened), I would still need to check if a specific error had been recorded surely? And gather all the userIDs with that error. That's why I felt like it made the most sense to record the 2 errors I'm interested in (email send failure and no user found) and store a list of userIds that I can then grab for the relevant response message

Reply:
>I would still need to check if a specific error had been recorded surely?

Yeah, but you can still do that with the way I suggested.

If you want to keep this like this and also prefer it to extracting it to its own class, that's fine, but you should change some things, because it really is too noisy and distracting. For instance, have a look at this code:

```java
      if (!failedUpdates.containsKey(FAILED_TO_SEND)) {
        List<Long> userList = new ArrayList<>();
        userList.add(userId);
        failedUpdates.put(FAILED_TO_SEND, userList);
      } else {
        List<Long> userList = failedUpdates.get(FAILED_TO_SEND);
        userList.add(userId);
      }
```

This is not only duplicated code, but it's 8 lines of code when all you really want is to record that a certain error happened when dealing with a certain user.

One way to make this easier to read is to initialise both `USERS_NOT_FOUND` and `FAILED_TO_SEND` to empty lists (or probably sets). For instance:

```java
    Map<String, Set<Long>> failedUpdates = new HashMap<>();
    failedUpdates.put(USERS_NOT_FOUND, new HashSet<>());
    failedUpdates.put(FAILED_TO_SEND, new HashSet<>());
```

Now you know that both keys are always present and initialised. Then those 8 lines simply become:

```java
failedUpdates.get(FAILED_TO_SEND).add(userId);
```

Of course, it also means that `failedUpdates.containsKey(FAILED_TO_SEND)` becomes `!failedUpdates.get(FAILED_TO_SEND).isEmpty()`, but that's not a big deal.

Reply:
Ok yeah that's much better